eng/helix/content/RunTests/TestRunner.cs

```diff
@@ -91,18 +92,26 @@ public async Task<bool> InstallAspNetAppIfNeededAsync()
         {
             try 
             {
-                Console.WriteLine("Checking for Microsoft.AspNetCore.App/");
-                if (Directory.Exists("Microsoft.AspNetCore.App"))
+                if (File.Exists(Options.AspNetRuntime))
                 {
                     var appRuntimePath = $"{Options.DotnetRoot}/shared/Microsoft.AspNetCore.App/{Options.RuntimeVersion}";
                     Console.WriteLine($"Creating directory: {appRuntimePath}");
                     Directory.CreateDirectory(appRuntimePath);
-                    Console.WriteLine($"Found Microsoft.AspNetCore.App/, copying to {appRuntimePath}");
                     Console.WriteLine($"Set ASPNET_RUNTIME_PATH: {appRuntimePath}");
                     EnvironmentVariables.Add("ASPNET_RUNTIME_PATH", appRuntimePath);
-                    foreach (var file in Directory.EnumerateFiles("Microsoft.AspNetCore.App", "*.*", SearchOption.AllDirectories))
+                    Console.WriteLine($"Found AspNetRuntime: {Options.AspNetRuntime}, extracting *.txt,json,dll to {appRuntimePath}");
+                    using (var archive = ZipFile.OpenRead(Options.AspNetRuntime))
                     {
-                        File.Copy(file, Path.Combine(appRuntimePath, Path.GetFileName(file)), overwrite: true);
+                        foreach (var entry in archive.Entries)
+                        {
+                            // These are the only extensions that end up in the shared fx directory
```

Comment:
Is it important to filter based on extension?

Reply:
We could just flatten and dump all the files I guess, I mostly was just trying to mimic what the final contents of the shared framework directory has.  There's a bunch of other metadata in other folders that aren't really relevant.  This was a little cleaner than filtering for relative paths since we'd have to build those based on target framework and other things